x-pack/legacy/plugins/maps/public/layers/sources/es_search_source/es_search_source.js

```diff
@@ -100,19 +110,15 @@ export class ESSearchSource extends AbstractESSource {
 
   async getDateFields() {
     try {
-      const indexPattern = await this._getIndexPattern();
+      const indexPattern = await this.getIndexPattern();
       return indexPattern.fields.getByType('date').map(field => {
-        return { name: field.name, label: field.name };
+        return this.createField({ fieldName: field.name });
       });
     } catch (error) {
       return [];
     }
   }
 
-  getMetricFields() {
```

Comment:
Should not remove this method. Used by es_source. getFieldFormatter. Without it, you get the following error for es_search_source

<img width="300" alt="Screen Shot 2019-11-20 at 8 18 27 AM" src="https://user-images.githubusercontent.com/373691/69251398-8d1cbb80-0b6e-11ea-88a1-ed2ef6426334.png">